packages/menu/src/SubMenu.tsx

```diff
@@ -0,0 +1,349 @@
+`;
+
+type Glyph = keyof typeof glyphs;
+
+interface SharedSubMenuProps {
+  open?: boolean;
+  setOpen?: React.Dispatch<React.SetStateAction<boolean>>;
+  className?: string;
+  description: React.ReactElement;
+  disabled?: boolean;
+  active?: boolean;
+  glyph?: Glyph;
+  onExited?: ExitHandler;
+}
+
+interface LinkSubMenuProps extends HTMLElementProps<'a'>, SharedSubMenuProps {
+  href: string;
+}
+
+interface ButtonSubMenuProps
+  extends HTMLElementProps<'button'>,
+    SharedSubMenuProps {
+  href?: null;
+}
+
+type SubMenuProps = LinkSubMenuProps | ButtonSubMenuProps;
+
+function usesLinkElement(
+  props: LinkSubMenuProps | ButtonSubMenuProps,
+): props is LinkSubMenuProps {
+  return props.href != null;
+}
+
+const subMenuItemHeight = 36;
+
+const SubMenu = React.forwardRef((props: SubMenuProps, ref) => {
+  const {
+    title,
+    description,
+    href,
+    children,
+    setOpen,
+    onKeyDown,
+    className,
+    onClick,
+    glyph,
+    onExited = () => {},
+    open = false,
+    active = false,
+    disabled = false,
+    ...rest
+  } = props;
+
+  const { usingKeyboard: showFocus } = useUsingKeyboardContext();
+
+  const iconButtonRef: React.RefObject<HTMLElement | null> = useRef(null);
+
+  const onRootClick = (
+    e: React.MouseEvent<HTMLAnchorElement, MouseEvent> &
+      React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    if (iconButtonRef.current?.contains(e.target as HTMLElement)) {
+      e.preventDefault();
+    } else if (onClick) {
+      onClick(e);
+    }
+  };
+
+  const renderedSubMenuItem = (Root: React.ElementType<any> = 'button') => (
+    <li role="none" className={liStyle}>
+      <Root
+        {...subMenuContainer.prop}
+        {...rest}
+        onKeyDown={onKeyDown}
+        role="menuitem"
+        href={href}
+        aria-haspopup="true"
+        ref={ref as RefObject<any>}
+        onClick={onRootClick}
+        className={cx(
+          menuItemContainerStyle,
+          subMenuStyle,
+          linkStyle,
+          {
+            [activeMenuItemContainerStyle]: active,
+            [disabledMenuItemContainerStyle]: disabled,
+            [subMenuOpenStyle]: open,
+            [focusedMenuItemContainerStyle]: showFocus,
+          },
+          className,
+        )}
+      >
+        {glyph && (
+          <Icon
+            glyph={glyph}
+            size="xlarge"
+            className={cx(mainIconStyle, {
+              [activeIconStyle]: active,
+              [mainIconFocusedStyle]: showFocus,
+            })}
+          />
+        )}
+
+        <div>
+          <div
+            className={cx(mainTextStyle, {
+              [activeMainTextStyle]: active,
+            })}
+          >
+            {title}
+          </div>
+
+          <div className={cx({ [disabledTextStyle]: disabled })}>
+            {description}
+          </div>
+        </div>
+      </Root>
+      <IconButton
+        ref={iconButtonRef}
+        ariaLabel={open ? 'Caret Up' : 'Caret Down'}
```

Comment:
Wouldn't it be better for accessibility to set the label to the button's action?
```
ariaLabel={open ? 'Close Sub-menu' : 'Open Sub-menu'}
```

Reply:
Good point